src/crimson/osd/main.cc

```diff
@@ -66,6 +68,35 @@ auto partition_args(seastar::app_template& app, char** argv_begin, char** argv_e
   return make_pair(std::move(ceph_args), std::move(app_args));
 }
 
+bool set_crimson_daemon(std::vector<const char*>& ceph_args, Preforker& forker){
+  bool crimson_daemonize = true;
+  for (std::vector<const char*>::iterator i = ceph_args.begin(); i != ceph_args.end();++i) {
+    if (ceph_argparse_flag(ceph_args, i, "--foreground", "-f", (char*)NULL)){
+      crimson_daemonize = false;
+      break;
+    }
+  }
+ 
+  std::string err_msg;
+  int err = forker.prefork(err_msg);
```

Comment:
do we need to setup the pipe and wait on the child if we don't want to daemonize?

Reply:
so what the parent process do after we fork the child when not do daemonize?

Reply:
if we don't daemonize the service, there is no need to fork, IMO.

Reply:
oh, I misunderstand foreground, then that is easy.